lightning/src/routing/router.rs

```diff
@@ -784,6 +785,13 @@ where L::Target: Logger {
 					// Verify the liquidity offered by this channel complies to the minimal contribution.
 					let contributes_sufficient_value = available_value_contribution_msat >= minimal_value_contribution_msat;
 
+
+					// Do not consider candidates that exceed the MAX_TOTAL_CLTV_DELTA limit.
+					const MAX_TOTAL_CLTV_DELTA: u32 = 1000;
```

Comment:
1008 is a nice-round-week, but maybe lets make it configurable. I could totally use-cases for lightning that need longer timeouts, and also people not want to wait a week. I guess we can shove it through the `Payee` thing?

Reply:
Great! But it seem to me that the limit doesn't have anything to do with the `Payee` side of things, but is chosen/enforced by the payer? So probably `RouteParameters` or a user config option makes more sense?

Reply:
Oh, right, forgot we had a `RouteParameters` struct, yea, put it there :)

Reply:
Also, instead of changing every test, if we make it configurable we could just disable it for testing, not that it matters a ton either way.

Reply:
I now added a `max_total_cltv_expiry_delta` field to `RouteParameters` in order to make the limit configurable and handed it down to `get_route` in the same way that the other params are provided (see https://github.com/lightningdevkit/rust-lightning/pull/1234/commits/beb91f7f8093fa672851d03908ea92e0dcfb6e5f). However, this required to touch a lot of places, most of them being tests. Not sure if a global user config would not be the cleaner way to it.

Reply:
Yea, the `get_route` is a bit of a compatibility wrapper, we could switch to using the public interface in tests, but the test diff isn't all that bad right now IMO so no big deal.

Reply:
Great!